### PR TITLE
some code refactoring

### PR DESCRIPTION
**Pull request by scalder27**
When I was reading the code for the first time it was a little hard to
reason about, so I would like to suggest some small code improvements
(maybe):
1) ditched one small piece of code duplication
2) splitted a long code, that was setting indexes for node, to a bunch
of smaller and a bit more understandable pieces

### CODE REVIEW

**Comment by hudochenkov** [comment]
Wow, this is a huge change. I need time to understand what is going on in code now. I will try to read and understand all these changes tomorrow.

**Comment by scalder27** [comment]
Hi! It`s ok! Maybe I overdid a little =)

File: index.js [hunk 1]
```diff
@@ -2,7 +2,7 @@ var postcss = require('postcss');
 var path = require('path');
 var fs = require('fs');
 
-function getSortOrder(options) {
+function getSortOrderFromOptions(options) {
```
Comment (hudochenkov) [inline index.js]: I see no reason to rename this function.

File: index.js [hunk 2]
```diff
@@ -26,28 +26,33 @@ function getSortOrder(options) {
...
                     prop: propIndex
                 };
-            });
-        });
+
+                return innerOrder;
+            }, order);
+        }, {});
+}
+
```
Comment (hudochenkov) [inline index.js]: I'll better stick to my code in this part. This `reduce` thing and closures are very hard to understand, and the code becomes less maintainable and straightforward.

File: index.js (hunk 1 above)
Comment (hudochenkov) [inline index.js]: Oh, I see you use `FromOptions` suffix for another option. I'll reconsider my opinion about that. Sorry, I've read the code function by function :)

**Comment by hudochenkov** [comment]
Awesome job! A code now is more organized :) Please get back code for `getSortOrderFromOptions` I'm pointed out in a comment above ([this code](https://github.com/hudochenkov/postcss-sorting/blob/master/index.js#L29-L50)). After that I can merge this PR :)

**Comment by scalder27** [comment]
Great thanks for the review! Cool!
It`s understandable that reduce might be pretty hard to understand, but can we at least ditch reduce and normalize sortOrder and iterate that way:
```c#
// Add sorting indexes to order
var order = {};

(typeof sortOrder[0] === 'string' ? [sortOrder] : sortOrder)
    .forEach(function(group, groupIndex) {
        group.forEach(function(prop, propIndex) {
            order[prop] = {
                group: groupIndex,
                prop: propIndex
            };
        });
    });

return order;
```

**Comment by hudochenkov** [comment]
Yes, this looks good and understandable :)

**Comment by scalder27** [comment]
The rollback was done, and one single commit was delivered =)

**Comment by hudochenkov** [comment]
Thank you! :)

**Comment by scalder27** [comment]
I hope these changes will be helpfull for project and will make it a bit easier to develop new features, to fix issues. Thank you for active support!

**Pull request merged**